src/devices/GoPiGo3/GoPiGo3.cs

```diff
@@ -263,25 +268,11 @@ public string GetManufacturer()
             byte[] outArray = {SpiAddress, (byte)SpiMessageType.GetManufacturer,
                   0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
             byte[] reply = SpiTransferArray(outArray);
-            if (reply[3] != 0xA5)
+            if (reply[3] != CorrectData)
             {
                 throw new IOException("No SPI response");
             }
-            else
-            {
-                for (int ndx = 4; ndx < 24; ++ndx)
-                {
-                    if (reply[ndx] != 0)
-                    {
-                        retVal += (char)reply[ndx];
-                    }
-                    else
-                    {
-                        break;
-                    }
-                }
-            }
-            return retVal;
+            return Encoding.ASCII.GetString(reply.Skip(4).Where(c => c != 0).ToArray());
```

Comment:
Actually the `Where` part will not work correctly:
`65,66,67,0,1` -> `65,66,67,1`. I think this should be the GetString overload which takes offset and count and for finding 0 you can use:
https://docs.microsoft.com/en-us/dotnet/api/system.array.indexof?view=netcore-2.1#System_Array_IndexOf__1___0_____0_System_Int32_System_Int32_

Reply:
the returned data starts at index 4 and the rest of the buffer after the text is always 0. So this allow to just get the byte to be converted. Getting the first 0 will be longer. Just wondering why I should change this now